src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -56,8 +56,417 @@ private static unsafe void ThrowExceptionEncryptDecryptFail(string fullPath)
             throw Win32Marshal.GetExceptionForWin32Error(errorCode, fullPath);
         }
 
+        private static SafeFileHandle? TryOpenTargetNoAlternativeDataStream(
+            string lpFileName,
+            int dwDesiredAccess,
+            FileShare dwShareMode,
+            FileMode dwCreationDisposition,
+            int dwFlagsAndAttributes,
+            ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA fileInformation,
+            out string openedFileName)
+        {
+            // Default value for openedFileName.
+            openedFileName = lpFileName;
+
+            // Check the file isn't an alternative data stream
+            if (Path.GetFileName(lpFileName.AsSpan()).Contains(':'))
+            {
+                return null;
+            }
+
+            // Try to open it in a way where we can detect symlinks.
+            SafeFileHandle result = Interop.Kernel32.CreateFile(
+                lpFileName,
+                dwDesiredAccess,
+                dwShareMode,
+                dwCreationDisposition,
+                dwFlagsAndAttributes | Interop.Kernel32.FileOperations.FILE_FLAG_OPEN_REPARSE_POINT
+                );
+
+            // Check we successfully opened it.
+            if (result.IsInvalid)
+            {
+                return null;
+            }
+
+            // Read the file attributes.
+            if (!Interop.Kernel32.GetFileAttributesEx(lpFileName, Interop.Kernel32.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, ref fileInformation))
```

Comment:
should this all reuse https://github.com/dotnet/runtime/blob/0cf1f09d6a90d873e206a2357feafe5f05e56b43/src/libraries/Common/src/System/IO/FileSystem.Attributes.Windows.cs#L50

Reply:
hmm, I'm not sure (but probably), I hadn't seen `FillAttributeInfo` before. I will take a look into this at some point to work out what this means for my logic and if I need to use it